bin/convert_owl.py

```diff
@@ -0,0 +1,390 @@
+        os.makedirs(out_dir)
+    preds_flag = ' --include-all-predicates' if include_all_predicates else ''
+    command_str = f'runoak -i sqlite:{inpath} dump -o {out_path} -O fhirjson{preds_flag}'
+    _run_shell_command(command_str)
+    return out_path  # todo: When OAK changes to save multiple files, return out_dir
+
+
+def owl_to_fhir(
+    input_path_or_url: str, out_dir: str = OUTDIR, out_filename: str = None, include_all_predicates=False,
+    retain_intermediaries=False, intermediary_type=['obographs', 'semsql'][0], use_cached_intermediaries=False,
+    intermediary_outdir: str = None, convert_intermediaries_only=False
+) -> str:
+    """Run conversion"""
+    # Download if necessary & determine outpaths
+    intermediary_outdir = intermediary_outdir if intermediary_outdir else out_dir
+    input_path = input_path_or_url
+    url = None
+    maybe_url = urlparse(input_path_or_url)
+    if maybe_url.scheme and maybe_url.netloc:
+        url = input_path_or_url
+    if not out_filename and not url:
+        _id = '.'.join(os.path.basename(input_path).split('.')[0:-1])  # removes file extension
+        out_filename = f'CodeSystem-{_id}.json'
+    elif not out_filename and url:
+        out_filename = 'CodeSystem.json'
+    if url:
+        input_path = os.path.join(CACHE_DIR, out_filename.replace('.json', '.owl'))
+        download(url, input_path)
+
+    # Preprocessing: Special cases
+    if 'rxnorm' in input_path.lower() or 'rxnorm' in out_filename.lower():
+        input_path = _preprocess_rxnorm(input_path)
+
+    # Convert
+    if intermediary_type == 'obographs' or input_path.endswith('.ttl'):  # semsql only supports .owl
+        intermediary_path = owl_to_obograph(input_path, use_cached_intermediaries)
+        obograph_to_fhir(
+            inpath=intermediary_path,
+            out_dir=intermediary_outdir,
+            out_filename=out_filename,
+            include_all_predicates=include_all_predicates)
+    else:  # semsql
+        intermediary_path = owl_to_semsql(input_path, use_cached_intermediaries)
+        semsql_to_fhir(
+            inpath=intermediary_path,
+            out_dir=intermediary_outdir,
+            out_filename=out_filename,
+            include_all_predicates=include_all_predicates)
+    if convert_intermediaries_only:
+        return intermediary_path
+
+    # Cleanup
+    indir = os.path.dirname(input_path)
+    template_db_path = os.path.join(indir, '.template.db')
+    if os.path.exists(template_db_path):
+        os.remove(template_db_path)
+    if not retain_intermediaries:
+        # noinspection PyUnboundLocalVariable
+        os.remove(intermediary_path)
+        if intermediary_type == 'semsql':
+            # More semsql intermediaries
+            intermediary_filename = os.path.basename(intermediary_path)
+            os.remove(os.path.join(indir, intermediary_filename.replace('.db', '-relation-graph.tsv.gz')))
+    return os.path.join(out_dir, out_filename)
+
+
+def _run_favorites(
```

Comment:
@ShahimEssaid This way of converting 'favorites' is the only one that I wonder if you might have issue with. I can do this differently, but also I can remove it altogether and move it to my own personal script outside of VCS or in my own separate repo.